core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -130,6 +132,20 @@ class SparkHadoopUtil extends Logging {
     UserGroupInformation.loginUserFromKeytab(principalName, keytabFilename)
   }
 
+  def updateCurrentUserHDFSDelegationToken(): Unit = {
+    val conf = new Configuration()
+    val nsId = DFSUtil.getNamenodeNameServiceId(conf)
+    val isHaEnabled = HAUtil.isHAEnabled(conf, nsId)
+
+    if(isHaEnabled){
+      val ugi = UserGroupInformation.getCurrentUser
+      val uri = FileSystem.getDefaultUri(conf)
+      val map = DFSUtil.getHaNnRpcAddresses(conf)
```

Comment:
@steveloughran If this is not a good option, is there another way of doing this? Maybe look up a specific configuration? Considering the fact that the configuration params are not likely to change (since it would break compat), we could just directly use those?